yb-voyager/cmd/common.go

```diff
@@ -970,6 +970,83 @@ func storeTableListInMSR(tableList []sqlname.NameTuple) error {
 	return nil
 }
 
+var (
+	UNSUPPORTED_DATATYPE_XML_ISSUE = fmt.Sprintf("%s - xml", UNSUPPORTED_DATATYPE)
+	UNSUPPORTED_DATATYPE_XID_ISSUE = fmt.Sprintf("%s - xid", UNSUPPORTED_DATATYPE)
+	APP_CHANGES_MAX_THRESHOLD      = 5
+	APP_CHANGES_MIN_THRESHOLD      = 1
+	SCHEMA_CHANGES_MAX_THRESHOLD   = int(math.Inf(1))
+	SCHEMA_CHANGES_MIN_THRESHOLD   = 20
+)
+
+var appChanges = []string{
+	INHERITANCE_ISSUE_REASON,
+	CONVERSION_ISSUE_REASON,
+	DEFERRABLE_CONSTRAINT_ISSUE,
+	UNSUPPORTED_DATATYPE_XML_ISSUE,
+	UNSUPPORTED_DATATYPE_XID_ISSUE,
+	UNSUPPORTED_EXTENSION_ISSUE, // will confirm this
+}
+
+func parseToInt(s string, env string) int {
+	res, err := strconv.Atoi(s)
+	if err != nil {
+		utils.ErrExit("error parsing the env var - %s: %v", env, err)
+	}
+	return res
+}
+
+func readEnvForAppOrSchemaCounts() {
+	schemaCountMinStr := os.Getenv("SCHEMA_CHANGES_MIN_THRESHOLD")
+	schemaCountMaxStr := os.Getenv("SCHEMA_CHANGES_MAX_THRESHOLD")
+	appCountMinStr := os.Getenv("APP_CHANGES_MIN_THRESHOLD")
+	appCountMaxStr := os.Getenv("APP_CHANGES_MAX_THRESHOLD")
+	if schemaCountMaxStr != "" {
+		SCHEMA_CHANGES_MAX_THRESHOLD = parseToInt(schemaCountMaxStr, "SCHEMA_CHANGES_MIN_THRESHOLD")
+	}
+	if schemaCountMinStr != "" {
+		SCHEMA_CHANGES_MIN_THRESHOLD = parseToInt(schemaCountMinStr, "SCHEMA_CHANGES_MIN_THRESHOLD")
+	}
+	if appCountMaxStr != "" {
+		APP_CHANGES_MAX_THRESHOLD = parseToInt(appCountMaxStr, "APP_CHANGES_MAX_THRESHOLD")
+	}
+	if appCountMinStr != "" {
+		APP_CHANGES_MIN_THRESHOLD = parseToInt(appCountMinStr, "APP_CHANGES_MIN_THRESHOLD")
+	}
+}
+
+// Migration complexity calculation from the conversion issues
+func getMigrationComplexity(sourceDBType string, analysisReport utils.SchemaReport) string {
+	if sourceDBType != POSTGRESQL {
+		return "NOT AVAILABLE"
+	}
+	if analysisReport.SchemaSummary.MigrationComplexity != "" {
+		return analysisReport.SchemaSummary.MigrationComplexity
+	}
+	readEnvForAppOrSchemaCounts()
+
+	log.Infof("Calculating migration complexity..")
+	os.Getenv("APP_CHANGES_HIGH_")
+	appChangesCount := 0
+	for _, issue := range schemaAnalysisReport.Issues {
+		for _, appChange := range appChanges {
+			if strings.Contains(issue.Reason, appChange) {
+				appChangesCount++
+			}
+		}
+	}
+	schemaChangesCount := len(schemaAnalysisReport.Issues) - appChangesCount
+
+	if appChangesCount > APP_CHANGES_MAX_THRESHOLD || schemaChangesCount > SCHEMA_CHANGES_MAX_THRESHOLD {
```

Comment:
Let's call them 
- `APP_CHANGES_HIGH_THRESHOLD`
- `SCHEMA_CHANGES_HIGH_THRESHOLD`
- `APP_CHANGES_MEDIUM_THRESHOLD` 
-  `SCHEMA_CHANGES_MEDIUM_THRESHOLD`

then the logic can be:
```
if appChangesCount > APP_CHANGES_HIGH_THRESHOLD || schemaChangesCount > SCHEMA_CHANGES_HIGH_THRESHOLD {
else if appChangesCount > APP_CHANGES_MEDIUM_THRESHOLD || schemaChangesCount > SCHEMA_CHANGES_MEDIUM_THRESHOLD
else ...
```